src/index.ts

```diff
@@ -0,0 +1,144 @@
+require("dotenv").config();
+
+import discord, {
+  Message,
+  PartialMessage,
+  MessageReaction,
+  User
+} from "discord.js";
+
+import { logger, stdoutLog, channelLog } from "./features/log";
+// import codeblock from './features/codeblock';
+import qna from "./features/qna";
+import jobs from "./features/jobs";
+import autoban from "./features/autoban";
+import commands from "./features/commands";
+import setupStats from "./features/stats";
+import emojiMod from "./features/emojiMod";
+import { ChannelHandlers } from "./types";
+
+const bot = new discord.Client({
+  partials: ["MESSAGE", "CHANNEL", "REACTION"]
+});
+bot.login(process.env.DISCORD_HASH);
+
+export type ChannelHandlersById = {
+  [channelId: string]: ChannelHandlers[];
+};
+
+const channelHandlersById: ChannelHandlersById = {};
+
+const addHandler = (channelId: string, channelHandlers: ChannelHandlers) => {
+  channelHandlersById[channelId] = [
+    ...(channelHandlersById[channelId] || []),
+    channelHandlers
+  ];
+};
+
+const handleMessage = (msg: Message | PartialMessage) => {
+  if (msg.partial) {
+    return;
+  }
+
+  const channelId = msg.channel.id;
+
+  if (channelHandlersById[channelId]) {
+    channelHandlersById[channelId].forEach(channelHandlers => {
+      channelHandlers.handleMessage?.({ msg: msg as Message, bot });
+    });
+  }
+
+  if (channelHandlersById["*"]) {
+    channelHandlersById["*"].forEach(channelHandlers => {
+      channelHandlers.handleMessage?.({ msg: msg as Message, bot });
+    });
+  }
+};
+
+const handleReaction = (reaction: MessageReaction, user: User) => {
+  const channelId = reaction.message.channel.id;
+
+  if (channelHandlersById[channelId]) {
+    channelHandlersById[channelId].forEach(channelHandlers => {
+      channelHandlers.handleReaction?.({ reaction, user, bot });
+    });
+  }
+
+  if (channelHandlersById["*"]) {
+    channelHandlersById["*"].forEach(channelHandlers => {
+      channelHandlers.handleReaction?.({ reaction, user, bot });
+    });
+  }
+};
+
+logger.add(stdoutLog);
+logger.add(channelLog(bot, "479862475047567361"));
+
+// Amplitude metrics
+setupStats(bot);
+
+// reactiflux
+addHandler("103882387330457600", jobs);
+addHandler("106168778013822976", qna); // reactiflux-admin
+addHandler("193117606629081089", qna); // #q&a
+
+// btm server
+addHandler("479862475047567361", qna); // #general
+
+// common
+addHandler("*", commands);
+// addHandler('*', codeblock);
+addHandler("*", autoban);
+addHandler("*", emojiMod);
+
+bot.on("messageReactionAdd", async (reaction, user) => {
+  if (user.partial) {
```

Comment:
Why are we early returning here? Is there not a `.fetch()` on user we can call?

Reply:
Good catch! I added `fetch` calls here and on other places that were missing